text/0227-separate-semantic-conventions.md

```diff
@@ -0,0 +1,179 @@
+# Separate Semantic Conventions
+
+Move Semantic Conventions outside of the main Specifications and version them
+separately.
+
+## Motivation
+
+We need to allow semantic conventions to evolve mostly independent of the
+overall OpenTelemetry specification. Today, any breaking change in a semantic
+convention would require bumping the version number of the entirety of the
+OpenTelemetry specification.
+
+## Explanation
+
+A new github repository called `semantic-conventions` would be created in the
+OpenTelemetry organization.
+
+This would *initially* have the following structure:
+
+- Boilerplate files, e.g. `README.md`, `LICENSE`, `CODEOWNERS`, `CONTRIBUTING.md`
+- `Makefile` that allows automatic generation of documentation from model.
+- `semantic_conventions/` The set of YAML files that exist in
+  `{specification}/semantic_conventions` today.
+- `docs/` A new directory that contains human readable documentation for how to
+  create instrumentation compliant with semantic conventions.
+  - `resource/` - Contents of `{specification}/resource/semantic_conventions`
+  - `trace/` - Contents of `{specification}/trace/semantic_conventions`
+  - `metrics/` - Contents of `{specification}/metrics/semantic_conventions`
+  - `logs/`- Contents of `{specification}/logs/semantic_conventions`
+  - `schemas/` - A new location for [Telemetry Schemas](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/schemas/README.md)
+    to live. This directory will be hosted at
+    `https://opentelemetry.io/schemas/`
+
+Existing semantic conventions in the specification would be marked as
+moved, with documentation denoting the move, but preserving previous contents.
+
+Additionally, if the semantic conventions eventually move to domain-specific
+directory structure (e.g. `docs/{domain}/README.md`, with trace, metrics, events
+in the same file), then this can be refactored in the new repository, preserving
+git history.
+
+There will also be the following exceptions in the specification:
+
+- Semantic conventions used to implement API/SDK details will be fully specified in the `opentelemetry-specification` repo
+  and will not be allowed to change in the Semantic Convention directory.
+  - Error/Exception handling will remain in the specification.
+  - SDK configuration interaction w/ semantic convention will remain in the
+    specification. Specifically `service.name`.
+- The specification may elevate some semantic conventions as necessary for
+  compatibility requirements, e.g. `service.instance.id` and
+  [Prometheus Compatibility](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/compatibility/prometheus_and_openmetrics.md).
```

Comment:
On a related note, there's also the (experimental) Compatibility semantic conventions section, which only includes OpenTracing values at the moment: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/compatibility.md 

I'd vow to keep them in the Spec as well but I'm open to discuss it.

Reply:
If you want them in the Specification, I'd suggest you put them here:

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/compatibility/opentracing.md


We can still move the YAML files to the new repository and do a similar "specification owns these attributes and requires them in semconv" behavior we're using for things like `service.name`

@carlosalberto would you be able to put that PR to the specification together?  For now I'll assume the existing YAML/MD is being pulled to the semconv repository based on my proposal above.

Reply:
Works, will send a PR to move them to the opentracind.md doc.